lab_1/main.py

```diff
@@ -40,7 +70,14 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list or None:
     :param top_n: a number of the most common words
     :return: a list of the most common words
     """
-    pass
+    if not isinstance(freq_dict, dict):
+        return None
+    freq_dict = sorted(freq_dict.items(), key=lambda x: -x[1])
```

Comment:
what is it `-x[1]`?

Reply:
what happens here?

Reply:
what is it -x[1]?
It is a parameter by which the array will be sorted (the value of key ‘top’)
what happens here?
First we split freq_dict into a list of key-value pairs by using freq_dict.items(). Then we pass that list to a sorted function. Sorted is a function which is used to sort a list of object by a specific parameter (key). We pass that key to a function too. The type of the parameter is a lambda function which takes one argument and returns a value by which the list will be sorted. So we put an item of our key-value pairs list into lambda function and mark it as ‘x’ and return a value by which list will be sorted (the value of key ‘top’): x[1]